### ISSUE TITLE
Learning rate doesnt take in account ignored features

### ISSUE THREAD

**Issue by FedorTarasow** [catboost/catboost#2843]
### Problem Description
When using pool and marking some features as ignored. These ignored features are still influencing learning rate. I guess.

```Python
features=list(X.columns[:100])
Small_Pool = Pool(data=X[features], label=y,has_header=True)
model2 = CatBoostClassifier(iterations=1000,random_seed=1)
model2.fit(Small_Pool,logging_level='Silent')
model2.learning_rate_
```
return 0.020483000203967094

```Python
features=list(X.columns[:100])
pool_features=Big_Pool.get_feature_names()
model = CatBoostClassifier(ignored_features=list(set(pool_features)-set(features)),iterations=1000,random_seed=1)
model.fit(Big_Pool,logging_level='Silent')
model.learning_rate_
```
return Learning rate 0.06321100145578384 instead of 0.020483000203967094

Bug is represented here:
https://colab.research.google.com/drive/16Buo-5yaeeq7qOZb6D6L8kjcnWXA0c4n?authuser=1#scrollTo=HJvc_BbjqJk4

Problem is fixed if i use model.drop_unused_features() but it is rather unobvious.
